src/AWSLambda/Events/APIGateway.hs

```diff
@@ -45,15 +52,39 @@ data RequestIdentity = RequestIdentity
   , _riCognitoIdentityId             :: !(Maybe Text)
   , _riCaller                        :: !(Maybe Text)
   , _riApiKey                        :: !(Maybe Text)
-  , _riSourceIp                      :: !(Maybe Text)
+  , _riSourceIp                      :: !(Word8, Word8, Word8, Word8)
```

Comment:
You're correct, there's no source port as we're not actually talking TCP directly here.

The advantage of `Data.IP.IP` is you'll be able to just `read` it (with appropriate error handling, of course) instead of reinventing the parsing here.

Reply:
I think it should be `Data.IP.IP`.